modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/FileEntryContentDashboardItem.java

```diff
@@ -404,6 +396,32 @@ public boolean isViewable(HttpServletRequest httpServletRequest) {
 		);
 	}
 
+	private String _getCopyURL() {
+		return Optional.ofNullable(
+			ServiceContextThreadLocal.getServiceContext()
+		).map(
+			ServiceContext::getLiferayPortletRequest
+		).map(
+			portletRequest -> {
+				List<ContentDashboardItemAction> contentDashboardItemActions =
+					getContentDashboardItemActions(
+						_portal.getHttpServletRequest(portletRequest),
+						ContentDashboardItemAction.Type.PREVIEW);
+
+				if (!contentDashboardItemActions.isEmpty()) {
+					ContentDashboardItemAction contentDashboardItemAction =
+						contentDashboardItemActions.get(0);
+
+					return contentDashboardItemAction.getURL();
+				}
+
+				return null;
```

Comment:
Just wondering if those "null" can cause any NullPointerException in other parts of the code
Maybe return empty string is safer?

Reply:
A null value is translated to a no value, but the empty string is used as is the actual string is empty. Where do you think a null pointer exception can be thrown?

Reply:
just checked all places where is used and won't produce any exception.
Nicely done @cgoncas  ;)